packages/medusa/src/services/region.ts

```diff
@@ -204,13 +232,24 @@ class RegionService extends BaseService {
   }
 
   /**
-   * Validates fields for creation and updates. If the region already exisits
+   * Validates fields for creation and updates. If the region already exists
    * the id can be passed to check that country updates are allowed.
-   * @param {object} region - the region data to validate
-   * @param {string?} id - optional id of the region to check against
-   * @return {object} the validated region data
+   *
+   * @param region - the region data to validate
+   * @param id - optional id of the region to check against
+   * @return the validated region data
    */
-  async validateFields_(region, id = undefined) {
+  async validateFields_<T extends CreateRegionInput>(
+    region: ValidationPartial<T>
```

Comment:
suggestion: since the type is used here only, I think that we can safely use the Omit directly